tools/alphafill_cofactor_survey/find_cofactor.py

```diff
@@ -69,22 +72,116 @@ def load_filter_file(filepath):
+
 def main():
+
     parser = argparse.ArgumentParser(description="Query AlphaFill for cofactors, with optional filtering.")
     parser.add_argument("-u", help="Single UniProt ID")
     parser.add_argument("-i", help="Index file with dataset, should have column 'UniProt AC'")
     parser.add_argument("-f", "--filter", help="Optional file with valid cofactor IDs")
     parser.add_argument("-o", "--output", default="summary_output", help="Output file prefix")
+    parser.add_argument("-p", "--pdb", help="PDB file to analyze for ligands")
+    parser.add_argument("-s", "--start", type=int, help="Start residue of trimmed region")
+    parser.add_argument("-e", "--end", type=int, help="End residue of trimmed region")
+    parser.add_argument("-c", "--cofactors", help="JSON file with cofactor definitions")
 
     args = parser.parse_args()
 
+    cofactor_list = set()
+    if args.cofactors:
+        try:
+            with open(args.cofactors, 'r') as f:
+                cofactors_data = json.load(f)
+            cofactor_list = set(cofactor.upper() for cofactor in cofactors_data.keys())
+        except FileNotFoundError:
+            print(f"Warning: {args.cofactors} not found, using basic classification")
+            cofactor_list = set()
+
     if not args.u and not args.i:
         print("Error: Provide either -u or -i")
         sys.exit(1)
 
     if args.u and args.i:
         print("Error: Provide either -u or -i, not both.")
-        sys.exit(1)        
+        sys.exit(1)
+
+    pdb_results = None
+
+    if args.pdb:
+        if not os.path.exists(args.pdb):
+            print(f"Error: PDB file {args.pdb} not found")
+            sys.exit(1)
+        if args.start is None or args.end is None:
+            print("Error: -s and -e required when using -p")
+            sys.exit(1)
+
+        print(f"Analyzing PDB file: {args.pdb}")
+        print(f"Residue range: {args.start}-{args.end}")
+
+
+        parser = PDBParser(QUIET=True)
+        structure = parser.get_structure('protein', args.pdb)
+        model = structure[0]
+
+
+        protein_residues = []
+        ligands = []
+        for chain in model:
+            for residue in chain:
+                if residue.id[0] == ' ':
+                    res_num = residue.id[1]
+                    if args.start <= res_num <= args.end:
+                        protein_residues.append(residue)
+                elif residue.id[0] != 'W':
+                    ligands.append(residue)
+        pdb_results = []
+
+
+        for ligand in ligands:
+            min_distance, contacts = calculate_ligand_contacts(ligand, protein_residues)
+
+            if len(contacts) == 0:
+                category = "no contacts"
+            else:
+                category = "inside trim"
+
+            ligand_name = ligand.resname.upper()
+
+            if ligand_name in cofactor_list:
+                ligand_type = "cofactor"
+            elif ligand_name in ['NA', 'K', 'CL', 'CA', 'MG', 'ZN', 'FE']:
```

Comment:
I think we should provide a way to customize this list - otherwise it's difficult to add/remove ions

e.g. there could be another flag to input a file with a list of ion names, similar to what we do for the other cofactors